content/blogs/2024-03-14-duck-db.md

```diff
@@ -0,0 +1,51 @@
+
+### We need a control plane
+
+To manage various data sources, applications, and infrastructure, a robust control plane is essential. Kestra aligns with the concept of a control plane by providing a centralized, comprehensive orchestration layer for managing data workflows, infrastructure, and applications. It simplifies complex operations, enabling users to define, automate, and monitor processes across different environments. This orchestration layer acts as a control plane by offering visibility and control over various components, streamlining execution, and enhancing efficiency in data-driven ecosystems.
+
+## About DuckDB
+
+DuckDB is considered SQLite for Analytics. It’s an open-source embedded OLAP database that can run in-process rather than relying on the traditional client-server architecture. As with SQLite, there’s no need to install a database server to get started. Installing DuckDB instantly turns your laptop into an OLAP engine capable of aggregating large volumes of data at an impressive speed using just CLI and SQL. It’s especially useful for reading data from local files or objects stored in cloud storage buckets (e.g., parquet files on S3).
+
+This lightweight embedded database allows fast queries in virtually any environment with almost no setup. However, it doesn’t offer high concurrency or user management, and it doesn’t scale horizontally. That’s where MotherDuck can help.
+MotherDuck
+
+## Orchestrate your Lakehouse with kestra and DuckDB
+
+In our exploration of DuckDB's integration within Kestra environments, we've identified several layers of complexity that showcase the applications of DuckDB as the query engine for lakehouse architectures. From automating basic queries to implementing advanced data management and analytics, these levels reflect the practical scenarios and solutions our user community is actively deploying:
+
+- **Level 1:** focuses on automating queries and managing dependencies within the stack, utilizing DuckDB for query execution and an S3 bucket for storage, with Kestra orchestrating the workflow. This level does not address SQL query management, metadata management, or "big data" capabilities.
+
+- **Level 2:** advances by incorporating SQL query management through tools like SQLMesh or DBT, still orchestrated by Kestra. It enhances the structure from Level 1 but remains limited by the absence of metadata management and "big data" support.
+
+- **Level 3:** represents a fully-fledged lakehouse model, adding metadata management, ACID properties, and time travel features through the integration of Apache Iceberg for the transaction layer. This level maintains the advancements of Level 2 and addresses its limitations, offering a comprehensive solution for data management and analytics.
```

Comment:
```suggestion
- **Level 3:** represents a fully-fledged lakehouse model, adding metadata management, ACID properties, and time travel features through the integration of Apache Iceberg for the transaction layer. This level maintains the advancements of Level 2 and addresses its limitations, offering a comprehensive solution for data management and analytics. Note that this integration is still in its early phase: Iceberg writing disposition with DuckDB is still in development and several challenges could arise. Still, it's something open-source [developers are working on](https://github.com/duckdb/duckdb_iceberg) and it could be a very interesting and mature solution soon.
```